demo-app/src/main/java/org/openmined/syft/demo/datasource/LocalMNISTDataDataSource.kt

```diff
@@ -2,32 +2,65 @@ package org.openmined.syft.demo.datasource
 
 import android.content.res.Resources
 import org.openmined.syft.demo.R
+import org.openmined.syft.demo.domain.Batch
 import java.io.BufferedReader
 import java.io.InputStreamReader
+import java.security.InvalidKeyException
 
 class LocalMNISTDataDataSource constructor(
     private val resources: Resources
 ) {
-    fun loadData(): Pair<List<FloatArray>, List<Float>> {
-        val trainInput = arrayListOf<FloatArray>()
-        val labels = arrayListOf<Float>()
+    private var trainDataReader = returnNewReader()
+    private val oneHotMap = HashMap<Int, List<Float>>()
 
-        val x = resources.openRawResource(R.raw.train_small)
-        BufferedReader(InputStreamReader(x))
-                .forEachLine { line ->
+    init {
+        for (i in 0..9) {
+            oneHotMap[i] = List(10) { idx ->
+                if (idx == i)
+                    1.0f
+                else
+                    0.0f
+            }
+        }
+    }
+
+    private val featureSize = 784
+
+    fun loadDataBatch(batchSize: Int): Pair<Batch, Batch> {
+        val trainInput = arrayListOf<List<Float>>()
+        val labels = arrayListOf<List<Float>>()
+        for (idx in 0..batchSize) {
+            val sample: List<String>? = trainDataReader.readLine()?.split(',')
+            sample?.let { sampleList ->
                     trainInput.add(
-                        line.split(',')
-                                .map {
-                                    it.trim().toFloat() / 255
-                                }.toFloatArray()
+                        sampleList.slice(1..featureSize).map { it.trim().toFloat() }
                     )
-                }
+                oneHotMap[sampleList[0].toInt()]?.let {
+                    labels.add(it)
+                } ?: throw InvalidKeyException("key not found ${sampleList[0]}")
+            } ?: break
+        }
 
-        val y = resources.openRawResource(R.raw.labels_small)
-        BufferedReader(InputStreamReader(y))
-                .forEachLine { line ->
-                    labels.add(line.toFloat() / 10)
-                }
-        return Pair(trainInput, labels)
+        //restart buffered reader to start of file
+        if (labels.size == 0) {
```

Comment:
This allows us to have infinite batches. Once the reader reaches the end of the file, we need to restart the buffer so that we can keep loading batches.